src/test/regress/expected/multi_name_lengths.out

```diff
@@ -362,6 +376,23 @@ SELECT master_create_worker_shards(U&'elephant_!0441!043B!043E!043D!0441!043B!04
  
 (1 row)
 
+SELECT shard_name(U&'elephant_!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D' UESCAPE '!'::regclass, min(shardid))
+FROM pg_dist_shard
+WHERE logicalrelid = U&'elephant_!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D' UESCAPE '!'::regclass;
+                   shard_name                    
+-------------------------------------------------
+ elephant_слонслонслонсло_c8b737c2_2250000000002
+(1 row)
+
+-- Verify that quoting is used in shard_name
+SELECT qualified_shard_name(U&'elephant_!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D' UESCAPE '!'::regclass, min(shardid))
+FROM pg_dist_shard
+WHERE logicalrelid = U&'elephant_!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D!0441!043B!043E!043D' UESCAPE '!'::regclass;
+                   qualified_shard_name                   
+----------------------------------------------------------
+ public."elephant_слонслонслонсло_c8b737c2_2250000000002"
```

Comment:
Using UTF-8 in tests isn't generally a good idea, depends quite a bit on the current environment.